connectors-home/modules/ROOT/pages/introduction/intro-config-oauth2.adoc

```diff
@@ -0,0 +1,189 @@
++
+Ensure that the user running the flow has access to that server and endpoint, otherwise the authorization approval does not reach the connector.
++
+You can use `localhost` for testing locally, but you cannot use `localhost` for production environments and cloud deployment, for example, if your connector application is running on `\http://my.server.com`, you must specify the *External Callback Url* field with the `\http://my.server.com/callback` value. The `\http://my.server.com/callback` URL must also be registered with the OAuth provider.
+. Once the user's browser successfully reaches the callback URL, the connector obtains an authorization code that is automatically exchanged for an access token using the URL configured in the *Access Token URL* field. This token is long-lived and is automatically refreshed by the connector indefinitely (if the OAuth provider allows so).
+
+The connector is now fully authorized and can use any service that accepts its new access token issued by the OAuth provider.
+
+== Example: How to Configure OAuth 2.0 For Gmail Connector
+
+The following example shows how to configure OAuth 2.0 for Gmail Connector.
+
+=== Configuring the OAuth Provider
+
+To configure the OAuth provider for Gmail Connector, you can use Google Cloud Console. The following steps show how to configure the OAuth provider:
+
+NOTE: If you haven't enabled the Gmail API previously, you must create a new project and enable the Gmail API in the *Enabled APIs & Services* menu section.
+
+. Add a new client (your connector application) to Google OAuth 2.0. To do so, navigate to *APIs & Services > Credentials > Create Credentials > OAuth Client ID*.
+. Create a new *Web Application*.
+. Enter a name, for example, `OAuthDemoApp`.
+. Add the following URL to the *Authorized redirect URIs*: `\http://127.0.0.1:8080/callback`.
+. Click *Create* and copy the new client ID and client secret for later use.
+. Click *OK*.
+
+=== Configuring the Connector
+
+The following steps show how to configure the connector with OAuth 2.0:
+
+. Add the following dependency into your pom.xml.
++
+[source,xml,linenums]
+----
+<dependency>
+    <groupId>com.mulesoft.connectors</groupId>
+    <artifactId>mule4-gmail-connector</artifactId>
+    <version>1.0.5</version>
+    <classifier>mule-plugin</classifier>
+</dependency>
+----
+. Configure the OAuth 2.0 authorization code. Create a new Gmail Connector connection and use the client ID and the client secret from the previous section as your *Consumer Key* and *Consumer Secret* respectively. You can leave the *Resource Owner ID* field empty.
++
+NOTE: Gmail Connector comes with pre-filled values for *Authorization URL*, *Access Token URL*, and *Scopes*, so no further action is required. However, if you are setting up a different connector that uses a third-party OAuth provider such as Okta or Ping, you must use their respective *Authorization URL*, *Access Token URL*, and *Scopes*.
```

Comment:
Another nit of mine to make the writing more precise: "Gmail Connector comes with pre-filled values for the *Authorization URL*, *Access Token URL*, and *Scopes* fields,..."
Same comment globally when you are referring to fields.